flink-core/src/main/java/org/apache/flink/configuration/GlobalConfiguration.java

```diff
@@ -143,31 +155,54 @@ public static Configuration loadConfiguration(
         }
 
         // get Flink yaml configuration file
-        File yamlConfigFile = new File(confDirFile, LEGACY_FLINK_CONF_FILENAME);
         Configuration configuration;
-
-        if (!yamlConfigFile.exists()) {
-            yamlConfigFile = new File(confDirFile, FLINK_CONF_FILENAME);
+        if (loadLegacyConfigFile) {
+            File yamlConfigFile = new File(confDirFile, LEGACY_FLINK_CONF_FILENAME);
```

Comment:
I prefer that the `GlobalConfiguration` code no longer see nor operate on the legacy config file, just like mentioned in the comment above.